sycl/tools/aie-intrinsic/aie-intrinsic.cpp

```diff
@@ -0,0 +1,65 @@
+//==-- aie-intrinsic.cpp - collection of intrisics for the aie target ----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#define __AIE_RT__
+#include "aie-intrinsic.h"
+
+namespace aie_intr {
+
+// All the functions called here are intrinsic. They do not have symbols but are
+// just a function-like representation of AIE instruction, so they should not be
+// able to collide with another symbol
+
+int get_coreid() { return ::get_coreid(); }
+
+void memory_fence() { ::chess_memory_fence();  }
+void separator_scheduler() { ::chess_separator_scheduler(); }
+
+void acquire(unsigned id, unsigned val) { return ::acquire(id, val); }
+void release(unsigned id, unsigned val) { return ::release(id, val); }
+void acquire(unsigned id) { return ::acquire(id); }
+void release(unsigned id) { return ::release(id); }
+
+void core_done() { ::done(); }
+
+extern "C" void _Z13finish_kernelv();
+
+void soft_done() {
+  _Z13finish_kernelv();
+}
+
+void nop5() { ::nop(5); }
+
+uint32_t sread(int stream_idx) { return ::get_ss(stream_idx); }
```

Comment:
You have removed `device/vendor/xilinx/aie/intrinsics.cpp` but at least there were some comments which might be useful to other reader than you...

Reply:
cant we just say that the behavior is not hard to guess but if you cant, go look at aie/chess documentation to know the behavior ?

Reply:
OK. So pointing to the headers in Vitis could be useful in the top header of this file.

Reply:
what Vitis headers are you talking about ?
the comments are in `aie-intrinsic.h`